include/dali/core/cuda_error.h

```diff
@@ -119,11 +119,44 @@ inline void cudaResultCheck<CUresult>(CUresult status) {
     throw dali::CUDAError(status);
   }
 }
+
+template <typename Code>
+inline void cudaResultDtorCheck(Code code) {
+  static_assert(!std::is_same<Code, Code>::value,
+    "cudaResultDtorCheck not implemented for this type of status code");
```

Comment:
Why not:
```suggestion
  cudaResultCheck(code);
``` ?
This way we'd have a pattern that in destructors we use CUDA_DTOR_CALL and just some errors get better handling there.

Reply:
Good point